Automattic/vip
# Fix custom deploy zip file size limitation

## Description

This pull request replaces the `adm-zip` library with `node-stream-zip` to address a critical limitation: `adm-zip` does not support processing files larger than 2GB (https://github.com/cthackers/adm-zip/issues/334). As a result, our custom deploy validation fails when handling large zip files. The new library, `node-stream-zip`, provides robust support for large archives and offers a more efficient, stream-based approach to working with zip files.


## Pull request checklist


## New release checklist


## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Check out PR.
2. Run `npm run build`
3. Run `node ./dist/bin/vip-app-deploy-validate.js /tmp/valid-zip.zip`
